### ISSUE TITLE
Adhere to XDG specifications

### ISSUE THREAD

**Issue by PatrickDaG** [StreamController/StreamController#207]
**Describe the bug**
The apps puts all their data and config in `$HOME/.var`. This does no follow any specification, no other program uses such a directory.
Unfortunately this leads to a cluttered home directory and breaks setups that expect programs to put their data in a unified place.

**Expected Behaviour**
Ideally this program would follow the [XDG base directory spec](https://specifications.freedesktop.org/basedir-spec/0.8/).
This would mean:
- configuration such as the pages.json and static config should be put in `${XDG_CONFIG_HOME:-$HOME/.config}/streamcontroller`
- non configuration data such as plugin files should be put in `${XDG_DATA_HOME:-$HOME/.local/share}/streamcontroller`
- cached assets should be put in `${XDG_CACHE_HOME:-$HOME/.cache}/streamcontroller`

**Comment by Core447** [comment]
I agree that splitting the data dir into cache, config etc makes sense. The .var/com.... toplevel dir however, is the default and recommended dir for Flatpaks.

**Comment by oddlama** [comment]
There's seems to be a long standing proposal to change this in flatpak (https://github.com/flatpak/flatpak/issues/3997), so it also adheres to the XDG spec.

I guess at the end of the day you can freely choose what your application does, because I don't think there's anything in flatpak that would directly benefit from one or the other layout. Similar "switch to XDG" issues on other applications usually get a lot of support which leads me to believe that a lot of people would appreciate if it were using the XDG spec. Maybe you can have a backward compatible mode that continues to use `.var/com...` if it already exists?

**Comment by ImDevinC** [comment]
Closing this as we adhere to flatpak standards already and that is the recommended install method. If Flatpak changes their proposal, the app should pick it up then
